src/components/editPost/EditPostModal.jsx

```diff
@@ -0,0 +1,152 @@
+import React, { useEffect, useState } from "react";
+import Picker, { SKIN_TONE_NEUTRAL } from "emoji-picker-react";
+import styles from "../tweetBox/tweetBox.module.css";
+import { EmojiIcon, GifIcon, ImageIcon } from "assets/icons/icons";
+import placeholder from "assets/images/placeholder.png";
+import { useDispatch, useSelector } from "react-redux";
+import { editPost, getPost } from "redux/slices/postSlice";
+
+export const EditPostModal = ({
+  isEditPostModalOpen,
+  setIsEditPostModalOpen,
+  setIsListExpand,
+  postId,
+  text,
+  avatar,
+}) => {
+  const [showEmojiPicker, setShowEmojiPicker] = useState(false);
+  const [charCount, setCharCount] = useState(210 - parseInt(text.length));
+
+  const { user } = useSelector((state) => state.auth);
+
+  const [newTweetBoxContent, setNewTweetBoxContent] = useState(text);
+
+  const dispatch = useDispatch();
+
+  const onEmojiClick = (event, emojiObject) => {
+    setNewTweetBoxContent((prevState) => prevState + emojiObject.emoji);
+  };
+
+  const { currentPost, postActionLoading } = useSelector((state) => state.post);
+
+  const submitFormHandler = (e) => {
+    e.preventDefault();
+    dispatch(
+      editPost({
+        _id: postId,
+        firstName: user.firstName,
+        lastName: user.lastName,
+        userPhoto: user.userPhoto,
+        content: newTweetBoxContent,
+      })
+    );
+  };
+
+  return (
+    <div
+      id="myModal"
+      className="modal"
+      style={isEditPostModalOpen ? { display: "block" } : { display: "none" }}
+    >
+      <div className={`modal-content box-shadow ${styles.customModal}`}>
+        <div className="modal-body">
+          <form
+            className={`card-vertical signup-form ${styles.customModal}`}
+            onSubmit={(e) => {
+              submitFormHandler(e);
+              setShowEmojiPicker(false);
```

Comment:
suggestion: you can write setter function inside 'submitFormHandler' then point onclick event to it. If you are facing an issue of async behaviour of dispatch function then write setter function in 'then' block and attached 'then' block after dispatch function.